### ISSUE TITLE
jsonpatch

### ISSUE THREAD

**Issue by Arnold1** [evanphx/json-patch#91]
hi, 

i have two json struct like the following, which are equal (only the order of keys is shifted). why is  jsonpatch.Equal not returning true?

```
s1 := []byte(`"SelectedData":[{"ID": 5, "bla": "hello5"},{"ID": 1, "bla": "hello1"},{"ID": 2, "bla": "hello2"}]`)
s2 := []byte(`"SelectedData":[{"ID": 1, "bla": "hello1"},{"ID": 5, "bla": "hello5"},{"ID": 2, "bla": "hello2"}]`)
fmt.Println("jsonpatch.Equal:", jsonpatch.Equal(s1, s2))
```

if not how can I make it work?

**Comment by clagraff** [comment]
Hi. I took a quick [peak at the code](https://github.com/evanphx/json-patch/blob/master/patch.go#L233), and it looks like it when it is comparing two array values, it goes index by index. So the order would matter.

In your example for the first element in `SelectedData`, the `s1` object where `{"ID": 5 ...}` would be compared to `s2`'s `{"ID"; 1 ...}` which obviously wouldn't match up.

So while the unordered contents of `SelectedData` is the same between `s1` and `s2`, here the order matters.

**Comment by clagraff** [comment]
Hi again! I noticed that your example JSON is actually malformed.

You have:
```
"SelectedData":[{"ID": 5, "bla": "hello5"},{"ID": 1, "bla": "hello1"},{"ID": 2, "bla": "hello2"}]
```

but it should actually be:

```
{"SelectedData":[{"ID": 5, "bla": "hello5"},{"ID": 1, "bla": "hello1"},{"ID": 2, "bla": "hello2"}]}
```

Note the missing braces at the start and end.

In the event that you call `jsonpatch.Equal` with something that isnt _really_ json, it does a byte comparison. In which case your two strings (not _objects_) are entirely different.

That said, even if it were fixed, the ordering of objects in an array would still matter, and so `s1` and `s2` would still be considered unequal.
